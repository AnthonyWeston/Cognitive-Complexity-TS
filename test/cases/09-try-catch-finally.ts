function f() {
    try {
    } catch (e) {
    } finally {
    }
}

function g() {
    try {
        // depth 0
        while (true); // +1
    } catch (e) { // +1
        // depth 1
        while (true); // +1 +1
    } finally {
        // depth 0
        while (true); // +1
    }
}

function h() {
    try {
        while (true);
    } catch (e) {
    } finally {
    }
}

function i() {
    try {
    } catch (e) {
        while (true);
    } finally {
    }
}

function j() {
    try {
    } catch (e) {
    } finally {
        while (true);
    }
}

function k() {
    try {
    } catch (e) { // +1
        // depth 1
        try {
        } catch (e) { // +1 +1
        } finally {
        }
    } finally {
    }
}