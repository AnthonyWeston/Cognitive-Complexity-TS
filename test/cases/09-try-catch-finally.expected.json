{
    "09-try-catch-finally.ts": {
        "score": 16,
        "inner": [
            {
                "name": "f",
                "line": 1,
                "column": 1,
                "score": 1
            },
            {
                "name": "g",
                "line": 8,
                "column": 1,
                "score": 5
            },
            {
                "name": "h",
                "line": 21,
                "column": 1,
                "score": 2
            },
            {
                "name": "i",
                "line": 29,
                "column": 1,
                "score": 3
            },
            {
                "name": "j",
                "line": 37,
                "column": 1,
                "score": 2
            },
            {
                "name": "k",
                "line": 45,
                "column": 1,
                "score": 3
            }
        ]
    }
}